flutter/flutter-provider/state-management-in-flutter-using-provider-Part2.md

```diff
@@ -0,0 +1,209 @@
+```
+
+So, let's get started knowing more about provider
+
+### What are the types of provider
+
+&nbsp;
+
+Provider has enough categories to handle different type of objects
+
+The complete list of all the objects available is [here](https://pub.dev/documentation/provider/latest/provider/provider-library.html)
+
+| Name|Description|
+|-----|-----------|
+|Provider|The most basic form of provider. It takes a value and exposes it, whatever the value is.|
+|ListenableProvider|A specific provider for Listenable object. ListenableProvider will listen to the object and ask widgets which depend on it to rebuild whenever the listener is called.|
+|ChangeNotifierProvider|A specification of ListenableProvider for ChangeNotifier. It will automatically call `ChangeNotifier.dispose` when needed.|
+|ValueListenableProvider|Listen to a ValueListenable and only expose `ValueListenable.value`.|
+|StreamProvider|Listen to a Stream and expose the latest value emitted.|
+|FutureProvider|Takes a `Future` and updates dependents when the future completes.|
+
+&nbsp;
+
+### So, let's discover more about them
+
+### **Provider**
+
+This is a very basic form of `provider`, Provider that manages the lifecycle of the value it provides by delegating to a pair of `Create` and `Dispose`. This can provider any type of value.
+
+```dart
+class Counter {
+    int count = 0;
+
+  void increment() {
+      count += 1;
+      notifyListeners();
+  }
+}
+
+class Stateless extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Provider<Counter>(
+      create: (context) =>  Counter(),
+      dispose: (context, value) => value.dispose(),
+      child: ...,
+    );
+  }
+}
+
+```
+
+### **ListenableProvider** 
+
+ListenableProvider is a bit different from the other providers in the list, as this provider provides `state source` as the value.
+
+Need more clarity?
+ 
+> Sure, Other providers generally provide the single value from the `object<T>`, but `ListenableProvider` provides the whole model as the value to its dependents.
+When the state changes in Listenable, it notifies the provider about the state change. Now provider, via it's `InheritedWidget`
+marks it's listening dependent to rebuild.
+
+
+In the above example, the whole `Counter()` object is being returned from `create`, as this value is provided to the dependents, dependents use this value as `Counter()` and updates it's state and rebuilds again to show the updated value on the screen.
+
+### **ChangeNotifierProvider** 
+
+In change notifier provider everything is same as `ListenableProvider`, one additional specification of `ListenableProvider` for ChangeNotifier. It will automatically call `ChangeNotifier.dispose` when needed.
+
+ `ChangeNotifier` can be used with `ListenableProvider<T>`, but `ChangeNotifierProvider<T>` provides one additional benefit of disposing the ChangeNotifier when done using ChangeNotifier's dispose() method.
```

Comment:
@aatikakhan Is this info needed for a beginner for can we skip it? 
It will be better to explain the core provider in this article and keep advance stuff for part 3 maybe.